sdc/extensions/indexes/range_index_ext.py

```diff
@@ -0,0 +1,366 @@
+
+    name = types.none if name is None else name
+    is_named = False if name is types.none else True
+
+    def codegen(context, builder, sig, args):
+        data_val, name_val = args
+        # create series struct and store values
+        range_index = cgutils.create_struct_proxy(
+            sig.return_type)(context, builder)
+
+        range_index.data = data_val
+
+        if is_named:
+            if isinstance(name, types.StringLiteral):
+                range_index.name = numba.unicode.make_string_from_constant(
+                    context, builder, types.unicode_type, name.literal_value)
+            else:
+                range_index.name = name_val
+
+        if context.enable_nrt:
+            context.nrt.incref(builder, sig.args[0], data_val)
+            if is_named:
+                context.nrt.incref(builder, sig.args[1], name_val)
+
+        return range_index._getvalue()
+
+    ret_typ = RangeIndexType(is_named)
+    sig = signature(ret_typ, data, name)
+    return sig, codegen
+
+
+@sdc_overload(pd.RangeIndex)
+def pd_range_index_overload(start=None, stop=None, step=None, dtype=None, copy=False, name=None, fastpath=None):
+
+    _func_name = 'pd.RangeIndex().'
+    ty_checker = TypeChecker(_func_name)
+
+    if not (isinstance(copy, types.Omitted) or copy is False):
+        raise SDCLimitation(f"{_func_name} Unsupported parameter. Given 'copy': {copy}")
+
+    if not (isinstance(copy, types.Omitted) or fastpath is None):
+        raise SDCLimitation(f"{_func_name} Unsupported parameter. Given 'fastpath': {fastpath}")
+
+    dtype_is_np_int64 = dtype is types.NumberClass(types.int64)
+    if not _check_dtype_param_type(dtype):
+        ty_checker.raise_exc(dtype, 'int64 dtype', 'dtype')
+
+    # TODO: support ensure_python_int from pandas.core.dtype.common to handle integers as float params
+    if not (isinstance(start, (types.NoneType, types.Omitted, types.Integer)) or start is None):
+        ty_checker.raise_exc(start, 'number or none', 'start')
+    if not (isinstance(stop, (types.NoneType, types.Omitted, types.Integer)) or stop is None):
+        ty_checker.raise_exc(stop, 'number or none', 'stop')
+    if not (isinstance(step, (types.NoneType, types.Omitted, types.Integer)) or step is None):
+        ty_checker.raise_exc(step, 'number or none', 'step')
+
+    if not (isinstance(name, (types.NoneType, types.Omitted, types.StringLiteral, types.UnicodeType)) or name is None):
+        ty_checker.raise_exc(name, 'string or none', 'name')
+
+    if ((isinstance(start, (types.NoneType, types.Omitted)) or start is None)
+            and (isinstance(stop, (types.NoneType, types.Omitted)) or stop is None)
+            and (isinstance(step, (types.NoneType, types.Omitted)) or step is None)):
+        def pd_range_index_ctor_dummy_impl(
+                start=None, stop=None, step=None, dtype=None, copy=False, name=None, fastpath=None):
+            raise TypeError("RangeIndex(...) must be called with integers")
```

Comment:
Why not raise error at compile time?

Reply:
@AlexanderKalistratov  We would have different runtime behavior with pandas then. I'm not sure which way we should take - either comply with pandas as much as possible, or intercept at compile time errors like this that pandas could only catch at runtime, but have compilation failure instead of runtime error. What do you think?